addons/hr_expense/models/account_move_line.py

```diff
@@ -34,15 +34,26 @@ def _compute_tax_key(self):
                 line.tax_key = frozendict(**line.tax_key, expense_id=line.expense_id.id)
 
     def _compute_all_tax(self):
-        super()._compute_all_tax()
-        for line in self:
-            if line.expense_id:
-                for key in list(line.compute_all_tax.keys()):
-                    new_key = frozendict(**key, expense_id=line.expense_id.id)
-                    line.compute_all_tax[new_key] = line.compute_all_tax.pop(key)
+        expenses = self.filtered('expense_id')
+        super(AccountMoveLine, expenses.with_context(force_price_include=True))._compute_all_tax()
+        super(AccountMoveLine, self - expenses)._compute_all_tax()
+        for line in expenses:
+            for key in list(line.compute_all_tax.keys()):
+                new_key = frozendict(**key, expense_id=line.expense_id.id)
+                line.compute_all_tax[new_key] = line.compute_all_tax.pop(key)
+
+    def _compute_totals(self):
+        expenses = self.filtered('expense_id')
+        super(AccountMoveLine, expenses.with_context(force_price_include=True))._compute_totals()
+        super(AccountMoveLine, self - expenses)._compute_totals()
 
     def _compute_term_key(self):
         super()._compute_term_key()
         for line in self:
             if line.expense_id:
                 line.term_key = line.term_key and frozendict(**line.term_key, expense_id=line.expense_id.id)
+
+    def _convert_to_tax_base_line_dict(self):
+        if self.expense_id:
+            return super(AccountMoveLine, self.with_context(force_price_include=True))._convert_to_tax_base_line_dict()
+        return super()._convert_to_tax_base_line_dict()
```

Comment:
```suggestion
    def _convert_to_tax_base_line_dict(self):
        result = super()._convert_to_tax_base_line_dict()
        if self.move_id.expense_sheet_id:
            result.setdefault('extra_context', {})
            result['extra_context']['force_price_include'] = True
        return result
```
Then, your diff on `_compute_totals` can be removed

Reply:
Hum, if I remove _compute_totals, then Untaxed amount, Tax and Total are computed correctly. But Amount Due is not :grimacing: 

So I guess, I will leave it :)